python/ray/train/huggingface/huggingface_trainer.py

```diff
@@ -427,48 +426,42 @@ def _huggingface_train_loop_per_worker(config):
         train_torch_dataset, eval_torch_dataset, **config
     )
 
-    if trainer.args.push_to_hub and not trainer.args.hub_token:
-        warnings.warn(
-            "You have set `push_to_hub=True` but didn't specify `hub_token`. "
-            "Pushing to hub will most likely fail, as the credentials will not "
-            "be automatically propagated from the local enviroment to the Ray Actors. "
-            "If that happens, specify `hub_token` in `TrainingArguments`."
-        )
-
-    if trainer.args.evaluation_strategy in ("steps", IntervalStrategy.STEPS):
+    if (
+        trainer.args.logging_strategy != trainer.args.evaluation_strategy
+        and trainer.args.evaluation_strategy not in ("no", IntervalStrategy.NO)
+    ) or (
+        trainer.args.logging_strategy != trainer.args.save_strategy
+        and trainer.args.save_strategy not in ("no", IntervalStrategy.NO)
+    ):
```

Comment:
I think you can simplify this to 
```
strategies = [trainer.args.logging_strategy, trainer.args.evaluation_strategy, trainer.args.save_strategy]
strategies = [strategy for strategy in strategies if strategy not in ("no", IntervalStrategy.NO)]
if not all(strategy == strategies[0] for strategy in strategies):
    ...
```

Reply:
let me know what you think, but I think the above is easier to understand the logic.

Reply:
sounds good, let me do this